mgr/dashboard_v2: Added missing UndoTest test.

sebastian-philipp opened the pull request:
Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

**Edit**: 

This tests the models.send_command_api.undo_transaction
decorator.

Imported from standalone openATTIC.

Can be squashed with 58537981 .

LenzGr committed:
Initial commit of the openATTIC Manager module

Create the directory and basic module initialization.

Add a minimal `README.rst`

Signed-off-by: Lenz Grimmer <lgrimmer@suse.com>

Commit:
mgr/dashboard_v2: add initial frontend code

This contains the angular boilerplate, some initial structure and the navigation component.
Signed-off-by: Tiago Melo <tmelo@suse.com>

votdev committed:
Make host and port configurable.

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/openattic: ceph_module mock to support unit testing

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/openattic: added first unit test example

To run the tests do:

UNITTEST=true py.test --cov=. tests/

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/openattic: updated READMED with unit testing info

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/openattic: Fix cherrypy shutdown handler

This patch prevents cherrypy from crashing the mgr process when some
exception is thrown when shutting down cherrypy.

Signed-off-by: Ricardo Dias <rdias@suse.com>

votdev committed:
Add command to create/update username/password.

Signed-off-by: Volker Theile <vtheile@suse.com>

votdev committed:
Remove faulty semicolon.

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/openattic: Added pylint support and fixed linting errors

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/openattic: added tox conf file to run unit tests and linting

Signed-off-by: Ricardo Dias <rdias@suse.com>

votdev committed:
Rename the command to create/update the login credentials.

Signed-off-by: Volker Theile <vtheile@suse.com>

p-se committed:
Renamed dashboard

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

votdev committed:
Create proper text editor config files to maintain consistent coding styles.

Signed-off-by: Volker Theile <vtheile@suse.com>

ricardoasmarques committed:
mgr/dashboard_v2: Auth API

Signed-off-by: Ricardo Marques <rimarques@suse.com>

sebastian-philipp committed:
Added `RESTResource` Python class.

This is the base class for providing a RESTful interface to a resource.

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

rjfd committed:
msgr/dashboard_v2: Fix python 2/3 compability problems

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: fix lint warnings in restresource.py

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
msgr/dashboard_v2: auth module refactoring

Signed-off-by: Ricardo Dias <rdias@suse.com>

Commit:
mgr/dashboard_v2: Updated README

Signed-off-by: Lenz Grimmer <lgrimmer@suse.com>

rjfd committed:
mgr/dashboard_v2: pass json body keys as args to REST resource methods

Signed-off-by: Ricardo Dias <rdias@suse.com>

votdev committed:
* Replace double quotes with single quotes
* Log script_name on unauthorized access

Signed-off-by: Volker Theile <vtheile@suse.com>

sebastian-philipp committed:
Added test_restresource.py

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

sebastian-philipp committed:
Reduced timeout for the auth unit tests one second. Unit tests should be fast!

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

sebastian-philipp committed:
Fixed two small bugs regarding debugging of unit tests.

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

rjfd committed:
mgr/dashboard_v2: Controller annotations

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: Very simple ping example

Signed-off-by: Ricardo Dias <rdias@suse.com>

votdev committed:
Ignore the __pycache__ directory.

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/dashboard_v2: Added developer notes to README

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: Increased test_auth session_expire to avoid false negatives

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: Renamed/moved RESTResource to RESTController

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: fix AuthRequired usage typo in README

Signed-off-by: Ricardo Dias <rdias@suse.com>

ricardoasmarques committed:
mgr/dashboard_v2: Add frontend login/logout

Signed-off-by: Ricardo Marques <rimarques@suse.com>

votdev committed:
Adapt reviewer comments.

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/dashboard_v2: Refactoring of unit tests

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: Fix pylint executor and warnings/errors

Signed-off-by: Ricardo Dias <rdias@suse.com>

Commit:
mgr/dashboard_v2: Fix content type of JSON responses

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

sebastian-philipp committed:
Added RESTController test case

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

rjfd committed:
mgr/dashboard_v2: Combine coverage and tests results from py27 and py3

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: Unit tests refactoring

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: Moved set-login-cred command logic to Auth class

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: Fixed more pylint warnings

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: Added unit test for module.py

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: .gitignore update

Signed-off-by: Ricardo Dias <rdias@suse.com>

ricardoasmarques committed:
mgr/dashboard_v2: Rename angular components prefix

Signed-off-by: Ricardo Marques <rimarques@suse.com>

Commit:
mgr/dashboard_v2: Fix content type (body) of error page

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

votdev committed:
Disable CherryPi autoreload feature. It does not work, either with the below code to fix the error

2018-01-26 13:21:15.779 7fa454add700 -1 log_channel(cluster) log [ERR] : Unhandled exception from module 'dashboard_v2' while running on mgr.x: execv() arg 2 must not be empty
2018-01-26 13:21:15.779 7fa454add700 -1 dashboard_v2.serve: File "/ceph/src/pybind/mgr/dashboard_v2/module.py", line 112, in serve

# Get the CherryPy autoreload feature working.
def _get_true_argv():
    return sys.argv[:]
cherrypy.engine._get_true_argv = _get_true_argv

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/dashboard_v2/frontend: add support for CI coverage and unit testing

Signed-off-by: Ricardo Dias <rdias@suse.com>

votdev committed:
- Replace double quotes with single quotes
- Modify log messages
- Fix typo, rename autenticate to authenticate

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/dashboard_v2: fix pycodestyle to exclude the frontend files

Signed-off-by: Ricardo Dias <rdias@suse.com>

ricardoasmarques committed:
mgr/dashboard_v2: Add hosts page

Signed-off-by: Ricardo Marques <rimarques@suse.com>

votdev committed:
mgr/dashboard_v2: Session expire at browser close

Add support for session expire at browser close.

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/dashboard_v2: removed auth handler form test_auth.py setup

The auth setup is already being set in module.py

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: fix linting error in module.py

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: Refactoring of the access to mgr module instance

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: Updated README with more developer notes

* Adds instructions on how to access the manager module instance from
  within the controller.
* Adds instructions on how to write unit tests to test a controller

Signed-off-by: Ricardo Dias <rdias@suse.com>

sebastian-philipp committed:
Imported Nodb classes

* Some fixes were needed for Nodb.

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

sebastian-philipp committed:
Added Cluster Nodb model

Lots of fixes were needed for CephCluster, but
unfortunately, we don't have that many CephCluster tests.

It's not yet usable. The main goal was to fix the Python unit tests.

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

sebastian-philipp committed:
Added very basic and not yet usable REST controller for CephCluster

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

sebastian-philipp committed:
Skipped AuthTest.test_session_expire because it is too slow in my opiinion.

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

Commit:
mgr/dashboard_v2: simplify router and fix navigation

Signed-off-by: Tiago Melo <tmelo@suse.com>

sebastian-philipp committed:
`models` folder: make pylint happy

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

rjfd committed:
mgr/dashboard_v2: added settings module

Signed-off-by: Ricardo Dias <rdias@suse.com>

sebastian-philipp committed:
Fixed some issues in the cluster api endpoint.

Also fixed issue caused by importing to much from `models.cluster`

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

rjfd committed:
mgr/dashboard_v2: Added instructions on how to add config settings to README

Signed-off-by: Ricardo Dias <rdias@suse.com>

votdev committed:
mgr/dashboard_v2: Refactor controller

Make it a little bit more OO. This way the mgr and logger properties are well designed and documented which should help contributors to understand the code.

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/dashboard_v2: Fix pylint errors exposed in python 2.7

Signed-off-by: Ricardo Dias <rdias@suse.com>

sebastian-philipp committed:
mgr/dashboard_v2: Added status route to cluster

* `RESTController`: added `@detail_route`
* Fixed `RequestHelper.assertJsonBody`

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

rjfd committed:
mgr/dashboard_v2: Fix bug in AuthRequired decorator

This PR fixes the fllowing bug:
When a subclass of `RESTController` was decorated with the `AuthRequired`
decorator, then all controllers that were also subclasses of
`RESTController` were automatically required authentication when accessing
them.

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2:Fix pylint import ordering error in tools.py

Signed-off-by: Ricardo Dias <rdias@suse.com>

ricardoasmarques committed:
mgr/dashboard_v2: Fix typo on method isLoginActive

Signed-off-by: Ricardo Marques <rimarques@suse.com>

ricardoasmarques committed:
mgr/dashboard_v2: Rename "host" module to "cluster"

Signed-off-by: Ricardo Marques <rimarques@suse.com>

ricardoasmarques committed:
mgr/dashboard_v2: Don't use localized config for auth

Signed-off-by: Ricardo Marques <rimarques@suse.com>

sebastian-philipp committed:
mgr/dashboard_v2: Refactored logging

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

ricardoasmarques committed:
mgr/dashboard_v2: Add block pools page

Signed-off-by: Ricardo Marques <rimarques@suse.com>

rjfd committed:
mgr/dashboard_v2: Notification queue

Signed-off-by: Ricardo Dias <rdias@suse.com>

votdev committed:
mgr/dashboard_v2: Use Awesome Bootstrap Checkbox.

Signed-off-by: Volker Theile <vtheile@suse.com>

votdev committed:
mgr/dashboard_v2: Mark text with the i18n attribute

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/dashboard_v2: Updated README with notification queue instructions

Signed-off-by: Ricardo Dias <rdias@suse.com>

Commit:
mgr/dashboard_v2: add health page and its dependencies

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: Add table component

The current implementation isn't finished yet, but it's a start.

The ngx-datatable is used as the core of the table component. You can
use the table to show data in a table. You can search, paginate, sort,
refresh the table contents. Multi selection is possible, the details
of the selected items will be given to the specified detail component.

What will be fixed soon?
* Enable the usage of buttons in the table header
* Enable details inline not beneath the table
* Pagination to use a input field to switch pages
* The columns to show can be checked and predefined
* The selection made by the user will be saved in the local storage

Signed-off-by: Stephan Müller <smueller@suse.com>
Signed-off-by: Tiago Melo <tmelo@suse.com>

sebastian-philipp committed:
mgr/dashboard_v2: Fix linting errors

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

rjfd committed:
mgr/dashboard_v2: uniformize logging calls (squash with 03abb41)

Signed-off-by: Ricardo Dias <rdias@suse.com>

sebastian-philipp committed:
mgr/dashboard_v2: Fixed Exception when loaded by mgr

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

sebastian-philipp committed:
mgr/dashboard_v2: Added missing UndoTest test.

This tests the models.send_command_api.undo_transaction
decorator.

Imported from standalone openATTIC.

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

ricardoasmarques committed:
mgr/dashboard_v2: Move hosts option to cluster menu

Signed-off-by: Ricardo Marques <rimarques@suse.com>

ricardoasmarques committed:
mgr/dashboard_v2: Add breadcrumb on Hosts page

Signed-off-by: Ricardo Marques <rimarques@suse.com>

rjfd committed:
mgr/dashboard_v2: mocking of rados and rbd modules

Signed-off-by: Ricardo Dias <rdias@suse.com>

Commit:
mgr/dashboard_v2: add missing codealizer recommended rules

Signed-off-by: Tiago Melo <tmelo@suse.com>

jecluis commented on src/pybind/mgr/dashboard_v2/tests/test_send_command_api.py:
typo: s/excption/exception/

jecluis reviewed (changes requested):
Mind telling us all, maybe in the form of a descriptive commit message, what this is trying to achieve, what is being tested, and what units benefit from said testing?

Also, minor nit: you have a typo.

Commit:
mgr/dashboard_v2: add tslint no-consecutive-blank-lines rule

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: add tslint import linting rule

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: add some eslint rules to tslint

Signed-off-by: Tiago Melo <tmelo@suse.com>

p-se committed:
mgr/dashboard_v2: Migrate RGW daemon REST resource

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

rjfd committed:
mgr/dashboard_v2: Updated README with instruction on how to mock ceph modules

Signed-off-by: Ricardo Dias <rdias@suse.com>

votdev committed:
mgr/dashboard_v2: Add RGW page.

Signed-off-by: Volker Theile <vtheile@suse.com>

Commit:
mgr/dashboard_v2: add style guide to the frontend README

Add a README section for style guide where it mentions how to organize
the typescript imports.

Signed-off-by: Tiago Melo <tmelo@suse.com>

votdev committed:
mgr/dashboard_v2: Add cdPasswordButton directive

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/dashboard_v2: clean rados connection on shutdown

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: adaptation of dashboard RemoteViewCache as a decorator

We implemented a decorator called `ViewCache` that implements a similar
functionality as the dashboard `RemoteViewCache` class.
We even copied part of the code and adapted to the current
infrastructure.

Signed-off-by: Ricardo Dias <rdias@suse.com>

Commit:
mgr/dashboard_v2: decrease max-line-length to 100

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: force whitespace in imports list

Signed-off-by: Tiago Melo <tmelo@suse.com>

ricardoasmarques committed:
mgg/dashboard_v2: table component style improvements

Signed-off-by: Ricardo Marques <rimarques@suse.com>

s0nea commented on src/pybind/mgr/dashboard_v2/tests/test_nodb.py:
Hey Sebastian,

I'm pretty sure there are reasons why this file belongs to the pull request but I don't see them currently. Could you please describe it shortly?

p-se committed:
mgr/dashboard_v2: Add support for nested controllers

The `ApiController` decorator is used to define a API endpoints.  This
change will enable to use `/` in paths, so that nested controllers can
be used.

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

votdev committed:
mgr/dashboard_v2: Let the pagination input field look nicer

Signed-off-by: Volker Theile <vtheile@suse.com>

rjfd committed:
mgr/dashboard_v2: Added coverage tracking to module code

This will allow to collect code coverage information when the module
is running within the ceph-mgr daemon.

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: add more debug message to tools.py

Signed-off-by: Ricardo Dias <rdias@suse.com>

rjfd committed:
mgr/dashboard_v2: removed mock classes, unit tests now run against real cluster

We changed tox.ini to run a vstart.sh cluster and run the unit tests
against that cluster. All unit tests code was adapted to make requests
to the real dashboard_v2 backend using the `requests` python library

Signed-off-by: Ricardo Dias <rdias@suse.com>

sebastian-philipp commented on src/pybind/mgr/dashboard_v2/tests/test_nodb.py:
I thought, no one would complain if I remove unused imports.

votdev committed:
mgr/dashboard_v2: Use datatable on Cluster/Hosts page.

Signed-off-by: Volker Theile <vtheile@suse.com>

jecluis commented:
@sebastian-philipp is this patch two distinct logical changes?

sebastian-philipp commented:
@jecluis then let's keep these imports

sebastian-philipp commented:
@s0nea, @jecluis : I just re-added these unused imports.

s0nea commented on src/pybind/mgr/dashboard_v2/tests/test_nodb.py:
Oh, don't get me wrong. I didn't want to complain (already approved your pull request) , it was just for my understanding. 

Okay, so you removed them, because they were unnecessary, great :)

rjfd committed:
mgr/dashboard_v2: Fix copy-webpack-plugin version

Signed-off-by: Ricardo Dias <rdias@suse.com>

Commit:
mgr/dashboard_v2: Fix linting errors

Supposed to be squashed with 5236e6168d763c8b40742c33ffb0ebe8e55ce6e8.

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

Commit:
mgr/dashboard_v2: add view cache component

This component should be used in components that access endpoints
that implement ViewCache.
It will inform the user if the data is stale or if there were some
problems fetching it.

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: adapt pool detail with cache view component

This modifies how pool detail show the cache information to the user
and implements a continuous update of the images information.

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: add 404 page

Add a 404 page that will be triggered every time the user tries to
access a route that does not exists or the servers responds with 404.

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: Update documentation

* Nested Controllers
* How to use RESTController

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

Commit:
mgr/dashboard_v2: Easy table cell transformation

The table column definition is extended by "cellTransformation"
property, which is filled through the use of "CellTemplate" enum.
If you have set that, a specific template reference will be used for
the ngx column property "cellTemplate". All references are stored
at the bottom of the data table template.

Signed-off-by: Stephan Müller <smueller@suse.com>

rjfd committed:
mgr/dashboard_v2: fix frontend bug in table.component.hml

Signed-off-by: Ricardo Dias <rdias@suse.com>

votdev committed:
mgr/dashboard_v2: Fix tox warnings.

Fix tox.ini to do not output the 'WARNING:test command found but not installed in testenv' message.

Signed-off-by: Volker Theile <vtheile@suse.com>

Commit:
mgr/dashboard_v2: add node as a local dependency

This will allow us to use the angular cli on systems that don't have node
v6.9.0 or higher installed.

Signed-off-by: Tiago Melo <tmelo@suse.com>

p-se committed:
mgr/dashboard_v2: Migrate standby module

This commit will add a StandbyModule class to enable redirects from any
inactive MGR instance to the active MGR instance.

Signed-off-by: Patrick Nawracay <pnawracay@suse.com>

votdev committed:
mgr/dashboard_v2: Highlight clickable datatable rows.

Signed-off-by: Volker Theile <vtheile@suse.com>

Commit:
mgr/dashboard_v2: add performance counter page

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: move datatable module

By moving the datatable module outside of the  components module we
can avoid a circular dependency between the datatable and sparkline
component.

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: add sparkline component

This also adds a template in the table component.

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: add monitors page

Signed-off-by: Tiago Melo <tmelo@suse.com>

Commit:
mgr/dashboard_v2: add router link table template

Signed-off-by: Tiago Melo <tmelo@suse.com>

rjfd committed:
mgr/dashboard_v2/frontend: fix visibility of class properties

Signed-off-by: Ricardo Dias <rdias@suse.com>

Commit:
mgr/dashboard_v2: Updated README.rst, added HACKING.rst

Moved developer-related documentation from `README.rst` to a
new file `HACKING.rst`.

Converted `frontend/README.md` to ReST and merged it into `HACKING.rst`.

Added updates to the README provided by @rjfd in PR#83
(updated the whole unit tests content to match the latest changes
on unit test execution, updated the development notes about
controller development)

Signed-off-by: Lenz Grimmer <lgrimmer@suse.com>

sebastian-philipp committed:
mgr/dashboard_v2: Renamed `block_pool` controller to `rbd`

* Simplified the API
* Fixed and improved the rbd controller tests.

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

Commit:
mgr/dashboard_v2: Add 'install' to HACKING.rst (squash with 82373ba)

'install' was missing from the 'pip install -r requirements.txt'
command on our HACKING.rst file. Running the command without 'install'
is not correct and leads to an error.

Signed-off-by: Tatjana Dehler <tdehler@suse.com>

sebastian-philipp commented:
For now, there won't be any NoDB in mgr/dashboard_v2.
